src/v/kafka/protocol/BUILD

```diff
@@ -0,0 +1,736 @@
+        "//src/v/kafka/protocol/schemata:end_txn_response",
+        "//src/v/kafka/protocol/schemata:fetch_request",
+        "//src/v/kafka/protocol/schemata:fetch_response",
+        "//src/v/kafka/protocol/schemata:find_coordinator_request",
+        "//src/v/kafka/protocol/schemata:find_coordinator_response",
+        "//src/v/kafka/protocol/schemata:heartbeat_request",
+        "//src/v/kafka/protocol/schemata:heartbeat_response",
+        "//src/v/kafka/protocol/schemata:incremental_alter_configs_request",
+        "//src/v/kafka/protocol/schemata:incremental_alter_configs_response",
+        "//src/v/kafka/protocol/schemata:init_producer_id_request",
+        "//src/v/kafka/protocol/schemata:init_producer_id_response",
+        "//src/v/kafka/protocol/schemata:join_group_request",
+        "//src/v/kafka/protocol/schemata:join_group_response",
+        "//src/v/kafka/protocol/schemata:leave_group_request",
+        "//src/v/kafka/protocol/schemata:leave_group_response",
+        "//src/v/kafka/protocol/schemata:list_groups_request",
+        "//src/v/kafka/protocol/schemata:list_groups_response",
+        "//src/v/kafka/protocol/schemata:list_offset_request",
+        "//src/v/kafka/protocol/schemata:list_offset_response",
+        "//src/v/kafka/protocol/schemata:list_partition_reassignments_request",
+        "//src/v/kafka/protocol/schemata:list_partition_reassignments_response",
+        "//src/v/kafka/protocol/schemata:list_transactions_request",
+        "//src/v/kafka/protocol/schemata:list_transactions_response",
+        "//src/v/kafka/protocol/schemata:metadata_request",
+        "//src/v/kafka/protocol/schemata:metadata_response",
+        "//src/v/kafka/protocol/schemata:offset_commit_request",
+        "//src/v/kafka/protocol/schemata:offset_commit_response",
+        "//src/v/kafka/protocol/schemata:offset_delete_request",
+        "//src/v/kafka/protocol/schemata:offset_delete_response",
+        "//src/v/kafka/protocol/schemata:offset_fetch_request",
+        "//src/v/kafka/protocol/schemata:offset_fetch_response",
+        "//src/v/kafka/protocol/schemata:offset_for_leader_epoch_request",
+        "//src/v/kafka/protocol/schemata:offset_for_leader_epoch_response",
+        "//src/v/kafka/protocol/schemata:produce_request",
+        "//src/v/kafka/protocol/schemata:produce_response",
+        "//src/v/kafka/protocol/schemata:sasl_authenticate_request",
+        "//src/v/kafka/protocol/schemata:sasl_authenticate_response",
+        "//src/v/kafka/protocol/schemata:sasl_handshake_request",
+        "//src/v/kafka/protocol/schemata:sasl_handshake_response",
+        "//src/v/kafka/protocol/schemata:sync_group_request",
+        "//src/v/kafka/protocol/schemata:sync_group_response",
+        "//src/v/kafka/protocol/schemata:txn_offset_commit_request",
+        "//src/v/kafka/protocol/schemata:txn_offset_commit_response",
+        "//src/v/utils:vint",
+        "@seastar",
+    ],
+)
+
+redpanda_cc_library(
```

Comment:
Worth a macro? 

I don't know if it's worth explicitly listing out each request response pair or not? Maybe in a bzl file we do something like:


```
HANDLERS = {
  "produce": [
    "product_request",
    "produce_response",
   ],
  ...
}
```

Then we can import it in multiple places and write a for comprehension. I am also fine if we only have one list and make sure everything is `${rpc}_request` and `${rpc}_response` and we just have a list of kafka RPCs.

Thoughts?

Reply:
ahh yeh good idea. i'll take a stab at something like this.

listing everything out explicitly is too verbose i agree. but having the fine grained libraries is nice because once we start getting into the server, we have different compilation units that only depend on indvidual request/response pairs.

Reply:
> but having the fine grained libraries is nice because once we start getting into the server

Totally, was not suggestion that we should combine them, fine-grained is the way! Just trying to reduce the noise in this file so people don't just blindly copy-paste later on (not the worst, but I think we can tidy it up).

Reply:
lemme know what you think about the approach in the appended commit.

Reply:
That's essentially what I was talking about - I like it! It's a lot smaller.